arches/app/models/migrations/10552_add_reference_datatype.py

```diff
@@ -0,0 +1,57 @@
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("models", "10541_controlled_list_manager"),
+    ]
+
+    operations = [
+        migrations.RunSQL(
+            sql="""
+            INSERT INTO d_data_types(
+                datatype,
+                iconclass,
+                modulename,
+                classname,
+                defaultconfig,
+                configcomponent,
+                configname,
+                isgeometric,
+                defaultwidget,
+                issearchable
+            ) VALUES (
+                'reference',
+                'fa fa-list',
+                'datatypes.py',
+                'ReferenceDataType',
+                '{"controlledList": null, "multiValue": false}',
+                'views/components/datatypes/reference',
+                'reference-datatype-config',
+                FALSE,
+                '19e56148-82b8-47eb-b66e-f6243639a1a8',
+                TRUE
+            );
+
+            INSERT INTO widgets(
+                widgetid,
+                name,
+                component,
+                datatype,
+                defaultconfig
+            ) VALUES (
+                '19e56148-82b8-47eb-b66e-f6243639a1a8',
+                'reference-select-widget',
+                'views/components/widgets/reference-select',
+                'reference',
+                '{"i18n_properties": ["placeholder"]}'
+            );
+
+            """,
+            reverse_sql="""
+            DELETE FROM widgets WHERE widgetid = '19e56148-82b8-47eb-b66e-f6243639a1a8';
+            DELETE FROM d_data_types WHERE datatype = 'reference';
+            """,
```

Comment:
Just noticed this failure trying to reverse migrate:

```py
django.db.utils.IntegrityError: update or delete on table "d_data_types" violates foreign key constraint "nodes_ddatatypes_fk" on table "nodes"
DETAIL:  Key (datatype)=(reference) is still referenced from table "nodes".
```

Reply:
This is pretty tricky. I pushed a fix that will allow reverse migration. I tried to clean out all the objects that might have used a node with the reference datatype, but in some cases it leaves some reference datatype cruft behind that I still have to track down.

Reply:
@chiatt for the reasons I laid out [here](https://github.com/archesproject/arches/pull/10641#discussion_r1512042566) you might consider not having a reverse migration at all.